scripts/client/publish-rust.mjs

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env zx
+import 'zx/globals';
+import { cliArguments, getCargo, workingDirectory } from '../utils.mjs';
+
+const dryRun = argv['dry-run'] ?? false;
+const [level] = cliArguments();
+if (!level) {
+  throw new Error('A version level — e.g. "path" — must be provided.');
+}
+
+// Go to the client directory and install the dependencies.
+cd(path.join(workingDirectory, 'clients', 'rust'));
+
+// Publish the new version.
+const releaseArgs = dryRun
+  ? []
+  : ['--no-push', '--no-tag', '--no-confirm', '--execute'];
+await $`cargo release ${level} ${releaseArgs}`;
```

Comment:
@febo Is it here that I need to set the `build` toolchain explicitly since releasing also builds the crate?

Reply:
Not sure if we need to build the release with a specific toolchain. In the end, the crate will be built but anyone that uses it as a dependency using their toolchain. It probably makes more sense to build with the “stable” one since it will likely be the one that most people will use.

Reply:
Cool, I'll leave it like that for now then. We can always revisit if needed.

Reply:
I think you should build it for release with a specified toolchain, only to make sure it doesn't fail. @febo is right that anyone consuming the crate will build it with their compiler version, so it would really just be to avoid a CI error. If you've been building with 1.75 all day long, and the build/test jobs in CI use 1.75, and then suddenly you merge and publish uses 1.72, and some dependency has an MSRV of 1.75, it will fail to publish.

Reply:
@febo is also correct, don't build and publish with nightly. I'm not even sure if Cargo will let you without forcing.

Reply:
Oh no the idea here is to have a new toolchain for building only. If that's `stable` by default then so be it.

Remember this?

```toml
[workspace.metadata.toolchains]
build = "1.75.0" <- We planned on adding this for building the Rust client.
format = "1.75.0"
lint = "1.75.0"
```

Reply:
I should probably read more of your changes before asking, but what's `cargo release`? The command I'm familiar with is `cargo publish`.

> Oh no the idea here is to have a new toolchain for building only. If that's stable by default then so be it.

I guess stable by default is the best we can do. My first comment was basically my long-winded way of saying there's a slight chance a CI publish job could fail if the toolchain is too new for publish, compared to the configured `build` version and/or MSRV of certain dependencies.

Reply:
If stable is what we want by default then let's just skip the `build` toolchain for now and keep things as-is then. We can always revisit in the future.

> what's cargo release? 

I think that's a helper crate that helps with bumping, building and publishing. This is what we used to use on previous templates to publish crates so I used it here as well. @febo might be able to provide more context.

Reply:
Nice, yeah that sounds good to me.